### ISSUE TITLE
Error: timeout of 0ms exceeded

### ISSUE THREAD

**Issue by adamreisnz** [axios/axios#2103]
Sentry reported a random error coming from Axios, `Error: timeout of 0ms exceeded`.

Under what circumstances would this error appear? We don't have timeout settings configured, and 0ms seems like a mighty short time for any request to finish, so why is this error thrown?

It appears to have been a one off that is not readily reproducible, so I can't provide test code, but the underlying network error seems to have been `ECONNABORTED`.

Axios version 0.18.0

**Comment by NatPDeveloper** [comment]
Getting same error in sentry after a post request.

Also just getting:

```sh
Error: Network Error
```

**Comment by jayrchamp** [comment]
@NatPDeveloper I don't know if these two are related : 

`Error: Network Error` 
& 
`Error: timeout of 0ms exceeded`

But you will find some hints about `Error: Network Error` here #383

**Comment by adamreisnz** [comment]
The Network Error is usually thrown when a browser can't connect to the internet or to the host. We don't report those in Sentry, but did get the 0ms exceeded errors in Sentry.

**Comment by vinnymac** [comment]
I am seeing this on an app hooked up with sentry and axios as well. For me this is only happening for users on macOS who are running the Safari browser.

Safari versions that have thrown the error:
12.1, 12.0.3, 12.0.2, 12.0, and 11.1.2

For my users they were all making a GET request. Same error message too ‘timeout of 0ms exceeded’.

From the breadcrumbs it appears that they all ran into this problem right as they were loading the app, as if they had a connectivity issue. The error code for the request seems to be ‘ECONNABORTED’. I wouldn’t expect Axios to be throwing an additional error here on top of the failed request.

**Comment by madisonbullard** [comment]
> I am seeing this on an app hooked up with sentry and axios as well. For me this is only happening for users on macOS who are running the Safari browser.
> 
> Safari versions that have thrown the error:
> 12.1, 12.0.3, 12.0.2, 12.0, and 11.1.2
> 
> For my users they were all making a GET request. Same error message too ‘timeout of 0ms exceeded’.
> 
> From the breadcrumbs it appears that they all ran into this problem right as they were loading the app, as if they had a connectivity issue. The error code for the request seems to be ‘ECONNABORTED’. I wouldn’t expect Axios to be throwing an additional error here on top of the failed request.

I am seeing this exact scenario in Sentry on a production app, except originating from iOS Safari. Here is user-agent:
```
Mozilla/5.0 (iPhone; CPU iPhone OS 11_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.0 Mobile/15E148 Safari/604.1
```
iOS 11.4.1
Mobile Safari 11.0

**Comment by qaifikhan** [comment]
I'm getting this exact same issue but in my case it's coming for all platforms - Windows, Linux, Mobile Browsers. Any clue what's happening hear?

**Comment by realfresh** [comment]
Same here, getting both errors on across all devices. Network error I understand but what exactly is timeout of 0ms exceeded? I looked through the source but can't see where this error is thrown.

**Comment by victorct-pulsarup** [comment]
Getting same error on Mac OS X and Google Chrome Version: 74.0.3729. Any ideas why or how can we fix this, it just started appearing.

**Comment by Undistraction** [comment]
Same error for Windows and Chrome 74.0.3729.

**Comment by JefferyHus** [comment]
Sentry is reporting `timeout of 0ms exceeded` & `Network Error`, this happens almost for every iOS safari user.

**Comment by JefferyHus** [comment]
My solution was to check either the `statusCode` or `Response` object

```javascript
      catch(function (error) {
        if (!error.response || error.code === 'ECONNABORTED') {
          // network error
        }
      })
```

**Comment by Undistraction** [comment]
We are seeing this when a stalled connection times out after 30 seconds. I think maybe the errors are valid, but the length of timeout is being misreported as 0.

**Comment by gpressutto5** [comment]
Getting same error here on these browsers:

- Chrome Mobile 74.0.3729
- Samsung Internet 9.2

**Comment by ovaldi** [comment]
+1

**Comment by qizhou23** [comment]
Getting same error here on these browsers:
- QQ Browser Mobile 6.2

**Comment by yingliangzhang** [comment]
Same here, it's happening for some of our iOS users reported by sentry, but I can't reproduce it myself.

sentry.javascript.browser
5.5.0

iOS
11.2.5

```
code: ECONNABORTED, 
column: 306412, 
config: {
adapter: [Function: <unknown-function-name>], 
headers: [Object], 
maxContentLength: -1, 
method: post, 
timeout: 0, 
transformRequest: [Object], 
transformResponse: [Object], 
validateStatus: [Function: validateStatus], 
xsrfCookieName: XSRF-TOKEN, 
xsrfHeaderName: X-XSRF-TOKEN
}, 
line: 1, 
message: timeout of 0ms exceeded, 
name: Error, 
request: {
__sentry_xhr__: [Object]
}, 
response: [undefined],
```

**Comment by chrisrickard** [comment]
Yup - i'm also getting Sentry errors of "timeout of 0ms exceeded" from iOS visitors only.

Does anyone know how this originates or the coarse (and whether it is "real"?)
I tried to replicate by connecting to an invalid server and then I get "Network Error" - I can't seem to recreate this as an actual timeout - but I am getting many users a day throwing the error.

**Comment by eisenest** [comment]
Any updates?

**Comment by laygir** [comment]
I just got this error from an Android device.

Chrome Mobile 76.0.3809 | SM-G960F

**Comment by navata** [comment]
I have same issue in react-native with Android. It have a error: "POST_ERROR order Error: timeout of 0ms exceeded".
Why ?

**Comment by ianokhin** [comment]
I am getting the same error `Error: timeout of 0ms exceeded` with `Error: Network Error` in bugsnag reports every day. Axios version 0.18.0. The majority of users affected by the issue use Chrome Mobile(v 76.0.3809). What may be the reasons for that? Why request timeout is 0ms?

**Comment by Undistraction** [comment]
For everyone enjoying this bug, I managed to catch this locally and match it up with a Sentry error report. It appears that this is the result of a genuine timeout, but the reporting of '0ms' is simply a bug in axios's reporting. So the errors are valid, however they are (or were in my case) real timeouts of significantly longer duration.

**Comment by jordyvandomselaar** [comment]
@Undistraction Is it possible to see the actual timeout?

**Comment by jordyvandomselaar** [comment]
@Undistraction did you do anything specific to reproduce this locally?

**Comment by malykhinvi** [comment]
We are also experience this problem, and we can not reproduce it. It is only visible in sentry logs. I'm doubt that it is bug in axios. It is clear though, that error is thrown here https://github.com/axios/axios/blob/6fe506fda290ba935c2641f68f1fcba7f4a16cd3/lib/adapters/xhr.js#L91
and `0ms` is displayed correctly, because it is 0 by default https://github.com/axios/axios/blob/1b07fb9365d38a1a8ce7427130bf9db8101daf09/lib/defaults.js#L71

I tried to find something similar on StackOverflow and on github and here is what I found:
- [Some timeout issue in jQuery](https://www.zohodiscussions.com/jquery/topic/jquery-ajax-gets-net-err-connection-timed-out-on-chrome-not-firefox-or-safari)
- [Chrome cache timeout issue](https://stackoverflow.com/questions/27513994/chrome-stalls-when-making-multiple-requests-to-same-resource?noredirect=1&lq=1)
- [Keep alive timeout](https://stackoverflow.com/questions/53414856/xhr-requests-fail-after-keepalivetimeout-has-elapsed-only-when-using-4g-connecti)
- [Some issue with persistent connection](https://stackoverflow.com/questions/27740692/request-stalled-for-a-long-time-occasionally-in-chrome)
- [Yet another issue with keep-alive](https://stackoverflow.com/questions/48227771/google-chrome-fails-to-load-xmlhttprequest-response-on-connection-close)

Trying to reproduce this issue I did multiple requests to the same endpoint, enabled ad blockers, did requests after connection close, turned off/on the internet, enabled throttling. Still no luck - I can not reproduce. 
Maybe the links I posted are somehow related to the problem.

One more detail: from logs I see that multiple api calls failed with "0ms timeout" simultaneously, even calls to different domains.

**Comment by shrpne** [comment]
@malykhinvi I believe this is an error caused by bad network connection and timed out by browser's inner timeout.

Axios here prints the wrong error message

**Comment by malykhinvi** [comment]
@Undistraction @shrpne do you guys identified steps to reproduce this issue?

**Comment by badz0** [comment]
Has anybody figured out what's the problem?
My users are getting the same errors (Chrome Mac OS, Chrome Linux).

**Comment by jordyvandomselaar** [comment]
In our case it was/is a combination of slow api calls and users losing connection, whether it be because they went from wifi to 4g or because they lost internet alltogether.

**Comment by adamreisnz** [comment]
@jasonsaayman why is this flagged as a question? I still think Axios should handle this properly somehow, a timeout of 0ms doesn't seem to make sense.

**Comment by shrpne** [comment]
@jasonsaayman @adamreisnz Agree, at least axios should change error message to something like "timeout exceeded" or "request timed out"

**Comment by jayrchamp** [comment]
@jordyvandomselaar 

> In our case it was/is a combination of slow api calls and users losing connection, whether it be because they went from wifi to 4g or because they lost internet alltogether.

This also seem to be our case. I would be curious to know how you handled these cases.

**Comment by jfdelarosa** [comment]
We're getting this error on every desktop OS, and because our product is a Chrome extension, only that browser is throwing the error.

We don't report bugs that are out of our hands, like "network error", but this one seems odd.

```
if (!SENTRY_IGNORED_ERRORS.includes(err.message)) {
  Sentry.captureException(err);
}
```

**Comment by sandeepg1220** [comment]
Getting Error: timeout of 0ms exceeded in android devices. But for iOS working fine. I'm using in react native.
Checked in Honor 8x, realme 3 pro devices. Tried in older version of axios 0.19.2 but same issue is getting.
It seems like old issue but still no solution. Any one can help for temporary solution for now. Due to this I was strucked.

Environment
Axios Version 0.21.1
React Native 0.62.1

**Comment by EmirhanAksoy** [comment]
I got the same error in iOS .

Axios Version 0.21.1

**Comment by severinbeauvais** [comment]
We see consistently in one app, right about the time the user is redirected to another app for authentication.

So, I wonder if this is because the axios call has not completed when the app is redirected, and the "force closed" (or "dangling" / "orphaned" / etc) axios connection is simply reporting the subject error, with no actual impact ?

**Comment by shrpne** [comment]
https://github.com/axios/axios/issues/2103#issuecomment-530516464 describes what happens here

In short: it is a network error caused by request timed out. Axios wrongly report 0ms timeout, when actually it should be something like "timeout of 60s exceeded"

**Comment by severinbeauvais** [comment]
> In short: it is a network error caused by request timed out. Axios wrongly report 0ms timeout, when actually it should be something like "timeout of 60s exceeded"

Well maybe I'm seeing a different problem with the exact same message though. I can tell from our Sentry logs that there was no 60s delay -- the axios error came milliseconds after the app started (followed by a redirect message).

**Comment by Tofandel** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/timeout
> The XMLHttpRequest.timeout property is an unsigned long representing the number of milliseconds a request can take before automatically being terminated. **The default value is 0, which means there is no timeout.**

So basically the issue is that some browsers will trigger the `ontimeout` callback when the connection is aborted, and it might not be the expected behavior

So I guess checking if the timeout was set as 0 in this callback to trigger a different error (the usual `Network Error`) for those  might be a good idea

**Comment by jasonsaayman** [comment]
Closed by #3209 this fixes the error with better wording

**Comment by Olimjon-create** [comment]
qande hal qilsa buladi
